text/0000-colocated-tests.md

```diff
@@ -0,0 +1,159 @@
+To counter this conflict, co-located tests must end with `.test`, e.g.: `my-component.test.js`. Because having extra periods in the filename isn't allowed in framework objects, using `.test` should be fairly safe to search for when building the app and test bundles.
+
+More concisely:
+
+For the app bundle,
+ - previously
+   - include: `app/**/*.{js,hbs}`
+ - now
+   - include: `app/**/*.{js,hbs}`
+   - exclude: `app/**/*.test.{js,ts}`
+
+For the test bundle,
+ - previously
+   - include: `tests/**/*.{js}`
+ - now
+   - include: `tests/**/*.{js}`
+   - include: `app/**/*.test.{js}`
+
+
+Like with component co-location, there will be different preferences with how to group files together. It may be undesirable to have a flat list of 20 routes, and then an additional flat list of 20 tests for those routes all in the same folder.
+The `.ember-cli` file provides the option to further nest components' class and template files as `app/components/{component-name}/index.{js,hbs}` via the `componentStructure` property (set to `nested`).
+This same flexibility should be available for routes, controllers, etc -- but is out of scope for this RFC as other resolver implementations may be in-flight.
+Avoiding implementing `componentStructure` for the rest of the `app` directory does not have any impact on test co-location as the resolver is not involved with finding test file locations. For example, it would be valid to have `app/routes/{route-name}.js` and tests located at `app/routes/__tests__/{route-name}.test.js`.
+
+
+### Reference: Where do the existing tests get migrated to?
+While tests in `app/` could live anywhere, these are to be the default locations for the generators to use.
+
+#### Route
+
+previously:
+ - `app/routes/{route-name}.js`
+ - `tests/unit/{route-name}-test.js`
+now:
+ - `app/routes/{route-name}.js`
+ - `app/routes/{route-name}.test.js`
+
+#### Service
+
+previously:
+ - `app/services/{service-name}.js`
+ - `tests/unit/{service-name}-test.js`
+now:
+ - `app/service/{service-name}.js`
+ - `app/service/{service-name}.test.js`
+
+#### Controller
+
+previously:
+ - `app/controllers/{controller-name}.js`
+ - `tests/unit/{controller-name}-test.js`
+now:
+ - `app/controllers/{controller-name}.js`
+ - `app/controllers/{controller-name}.test.js`
+
+#### Component
+
+previously:
+ - `app/components/{component-name}.js`
+ - `app/components/{component-name}.hbs`
+ - `tests/integration/components/{component-name}-test.js`
+now:
+ - `app/components/{component-name}.js`
+ - `app/components/{component-name}.hbs`
+ - `app/components/{component-name}.test.js`
+
+or using `"componentStructure": "nested"`:
+ - `app/components/{component-name}/index.js`
+ - `app/components/{component-name}/index.hbs`
+ - `app/components/{component-name}/.test.js`
```

Comment:
:thinking: What about `index.test.js` to prevent this file behave as a hidden on some systems?

Reply:
oh right! I forgot windows does that. good call. will update

Reply:
actually, I think Mac does, too.

and now that I think of it.. everyone does _by default_.

whoops